clusterman/kubernetes/kubernetes_cluster_connector.py

```diff
@@ -133,6 +134,62 @@ def freeze_agent(self, agent_id: str) -> None:
         except ApiException as e:
             logger.warning(f"Failed to freeze {agent_id}: {e}")
 
+    def drain_node(self, agent_id: str) -> bool:
+        try:
+            logger.info(f"Cordoning {agent_id}...")
+            if not self._cordon_node(agent_id):
+                return False
+            logger.info(f"Evicting pods on {agent_id}...")
+            if not self._evict_tasks_from_node(agent_id):
+                return False
+            logger.info(f"Drained  {agent_id}")
+            return True
+        except Exception as e:
+            logger.warning(f"Failed to drain {agent_id}: {e}")
+            return False
```

Comment:
I wonder if we should un-cordon the node if this fails half-ways, otherwise I feel like there may be the risk of ending up with a lot of "unschedulable" infra. Or maybe a revert after a certain number of failures.

Reply:
Good Idea. revert after certain number of tries or minutes++
I am trying to find best way.